### DISCUSSION TITLE
Allow adding root certificate

### DISCUSSION THREAD

**Discussion by M-Gregoire** [hassio-addons/addon-node-red#899] [Ideas]
# Problem/Motivation

I run my own PKI, all my services are exposed with HTTPS, were the certificate is signed by a root certificate authority. 
I'm trying to access some services from node-red (Inside a docker managed by HassOS) but the HTTPS requests fails with:
`"Error Message: unable to get local issuer certificate"`.

I wasn't able to find any way to add my root certificate so It can by accepted by node-red.
I also couldn't find any way to disable the SSL/TLS verification as a temporary fix. Modifying `settings.js` from inside the container being unpractical and not persistent between update.

## Proposed changes

I would love to see options such as:
- `verify_ssl: true`
- `authorized_ca: []`

which would allow greater flexibility.

I'm sorry if there is already a work-around I'm not aware of.

**Comment by frenck** [comment]
I'm not aware of a correct workaround for this, however, adding options for this is not something I would consider either.

The reasoning is that this simply is a use case that is pretty much edge/not widely used. Adding these options makes things more complicated for users and above all, adds maintenance over time.

**Reply by M-Gregoire** [reply]
I understand your reasoning,
Thank you for your quick answer.
